sdk/appcenter/src/main/java/com/microsoft/appcenter/ingestion/AppCenterIngestion.java

```diff
@@ -93,11 +106,24 @@ public void setLogUrl(@NonNull String logUrl) {
         mLogUrl = logUrl;
     }
 
+    @Override
+    public void setAuthToken(@NonNull String authToken) {
+        mAuthToken = authToken;
+    }
+
+    @Override
+    public String getAuthToken() {
+        return mAuthToken;
+    }
+
     @Override
     public ServiceCall sendAsync(String appSecret, UUID installId, LogContainer logContainer, final ServiceCallback serviceCallback) throws IllegalArgumentException {
         Map<String, String> headers = new HashMap<>();
         headers.put(INSTALL_ID, installId.toString());
         headers.put(APP_SECRET, appSecret);
+        if (mAuthToken != null && mAuthToken.length() > 0) {
```

Comment:
This block (or just the method) along with the getter and setters for auth token, should be synchronized because otherwise the auth token can become invalid inside this block from another thread

Reply:
If you add synchronized here, must also add to get and set.